tket2-py/test/test_guppy.py

```diff
@@ -9,8 +10,46 @@
 from guppylang.prelude.builtins import py
 from guppylang.prelude.quantum import measure, phased_x, qubit, rz, zz_max
 
+import pytket
 
-def test_load_compiled_module():
+
+def test_load_pure_circuit():
+    module = GuppyModule("test")
+    module.load(quantum)
+
+    @guppy(module)
+    @no_type_check
+    def my_func(
+        q0: qubit,
+        q1: qubit,
+    ) -> tuple[qubit, qubit]:  # pragma: no cover
+        q0 = phased_x(q0, py(math.pi / 2), py(-math.pi / 2))
+        q0 = rz(q0, py(math.pi))
+        q1 = phased_x(q1, py(math.pi / 2), py(-math.pi / 2))
+        q1 = rz(q1, py(math.pi))
+        q0, q1 = zz_max(q0, q1)
+        q0 = rz(q0, py(math.pi))
+        q1 = rz(q1, py(math.pi))
+        return (q0, q1)
+
+    # Compile the module
+    hugr = module.compile()
+    json = hugr.to_raw().to_json()
+
+    # Load it from the JSON string
+    circ = Tk2Circuit.from_guppy_json(json, "my_func")
```

Comment:
these three lines probably belong in a function, in _circuit.py orsomething

Reply:
There's a couple reasons for keeping this explicit for the moment:

Without https://github.com/CQCL/guppylang/issues/246, we need to keep an explicit reference to the hugr if we want to create multiple circuits from the same module.
Once that's published we'll be able to create functions from the function definition itself;
```python
from_guppy(my_func: guppy.Definition) -> Tk2Circuit
```

But more importantly, `tket2` only has a _dev-dependency_ on guppy.
To avoid cyclic dependencies, we cannot expose `from_guppy` in the library API.
In the future I imagine guppy may import tket2 and expose `to_circuit` methods on modules and functions.

Reply:
I added a utility functions for testing, but we won't be able to expose it in the tket2 api.

Reply:
makes sense, thanks for the explanation